js/donations-visualiser.js

```diff
@@ -125,6 +125,27 @@ d3.select("#receipt-type-select-all").on("click", function() { selectAll('#recei
 d3.select("#receipt-type-select-clear").on("click", function() { clearSelection('#receipt_type_select'); });
 d3.select("#clear-search").on("click", clearSearch);
 
+// on 'escape' key press, close the info window, and zoomToFit
+window.addEventListener("keydown", function (event) {
+  if (event.defaultPrevented) return;
+
+  var handled = false;
+  if (event.key === 27) {
+    handled = true;
+  } else if (event.keyIdentifier === 27) {
+    handled = true;
+  } else if (event.keyCode === 27) {
+    handled = true;
+  }
```

Comment:
Sounds good!
On 9 Jun 2016 6:52 PM, "Tim Osborn" <notifications@github.com> wrote:

> In js/donations-visualiser.js
> <https://github.com/river-jade/donations-visualiser/pull/16#discussion_r66403831>
> :
>
> > @@ -125,6 +125,27 @@ d3.select("#receipt-type-select-all").on("click", function() { selectAll('#recei
> >  d3.select("#receipt-type-select-clear").on("click", function() { clearSelection('#receipt_type_select'); });
> >  d3.select("#clear-search").on("click", clearSearch);
> >
> > +// on 'escape' key press, close the info window, and zoomToFit
> > +window.addEventListener("keydown", function (event) {
> > +  if (event.defaultPrevented) return;
> > +
> > +  var handled = false;
> > +  if (event.key === 27) {
> > +    handled = true;
> > +  } else if (event.keyIdentifier === 27) {
> > +    handled = true;
> > +  } else if (event.keyCode === 27) {
> > +    handled = true;
> > +  }
>
> yeah.. that code block was a dump from MDN, not my style.. hurried, wanted
> to clear quickly cos i was doing it over and over... the update uses
> jquery's event, which normalises key, keyCode etc.. stylewise, i agree.. if
> a line gets long, i like:
>
> if (
>   (longCondition && longCondition.someKey !== 'a random string')
>   || another condition
>   || aThirdOption
> ) {}
>
> // using leading operators, just like:
> const myPhrase = longCondition && longContition.smthg
>   ? 'true string'
>   : 'false string';
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/river-jade/donations-visualiser/pull/16/files/2a49d5a712e0d26812b77159fe82ef7ef30524fe..1ba2e76cc5fc7649a9bd51e5077e878f8947d34b#r66403831>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AArMkKOKpRt4Et_uDXXmYVdvgerIMAfnks5qJ9QygaJpZM4Ivg33>
> .
>